### DISCUSSION TITLE
How to use Velero in Vultr kubernetes Cluster to backup redis cluster

### DISCUSSION THREAD

**Discussion by Praveen-98cs** [vmware-tanzu/velero#7954] [Community support Q&A]
I have deployed redis-cluster version using bitnami helm chart in vultr kubernetes cluster, and for backup and restore they have suggested using velero. But when i check the documentation they haven't included any cloud agnostic way or in cloud providers, vultr. Vultr has Object storages where they are S3 compaitable can you help me with how to use velero in vultr to backup and restore redis-cluster with persistent storages? thank you!

**Comment by blackpiglet** [comment]
For the S3-compatible object storage, you can use https://github.com/vmware-tanzu/velero-plugin-for-aws to connect it.

Another thing worth mentioning is that the Velero backup will remove some metadata from the backed-up k8s resources. That may cause the restored resources is not adopted by the Helm.

**Reply by Praveen-98cs** [reply]
Thank you will update after trying out this!

**Reply by Praveen-98cs** [reply]
@blackpiglet I tried this method and it got the persistent storages when backed up and restored, but I couldn't retrieve the data in the snapshot file data ("dumb.rdb") which were situated in bitnami/redis/data location of each persistent storages. is there a way i have to define the location to backup the snapshot file to velero in some way? thanks in advance.

This is the helm command that i used to install velero in vultr kubernetes cluster (in velero namespace). I added the s3 key_id and access_key in the credentials-velero file 
```
velero install \
    --provider aws \
    --plugins velero/velero-plugin-for-aws:v1.2.0 \
    --bucket velero \
    --secret-file ./credentials-velero \
    --use-volume-snapshots=false \
    --backup-location-config region=ewr1,s3ForcePathStyle="true",s3Url=https://ewr1.vultrobjects.com \ 
    --use-node-agent \
    --namespace velero \
    --wait
```

**Comment by blackpiglet** [comment]
The snapshot backup you are using in this example is the Velero-Native snapshot.
It's provided with the cloud provider plugin. In this case, it's the AWS snapshot plugin, which is used to interact with the AWS snapshot API to manipulate the snapshots.

Because you are in an S3-compatible environment, not an AWS environment, the Velero-Native snapshot is available.

If Vultr provides CSI plugins for the storage, you can use the CSI way to back up the volume by the snapshot.
* [CSI](https://velero.io/docs/v1.14/csi/)
* [CSI snapshot data mover](https://velero.io/docs/v1.14/csi-snapshot-data-movement/)

If there are no CSI plugins, the only option is the filesystem backup/restore. https://velero.io/docs/v1.14/file-system-backup/

**Reply by Praveen-98cs** [reply]
Thank you will try this way and update
